tests/cuda2/dpct_output/Makefile

```
@@ -0,0 +1,28 @@
+CC=dpcpp
+CFLAGS=-O2 -Wall
+LDFLAGS=
+
+INCLUDES=-Iincludes
+
+all: main
+
+main: main.c dw_cuda_soa.dp.cpp read_in.c
+	$(CC) $(INCLUDES) $(CFLAGS) $^ -o $@
```

We will need to be able to use the Makefile to build with different compilers and architecture targets. I can use this one if I invoke it with

```
make CC=clang++ CFLAGS="-O2 -Wall -fsycl -fsycl-targets=nvptx64-nvidia-cuda"
```

but it feels a bit cumbersome. At the least we will need to document the commands we are using. Do you have any ideas how to improve it?

One way would be following [this stackoverflow thread](https://stackoverflow.com/questions/13358131/one-makefile-for-two-compilers) set `CC` and `CFLAGS` for each target and then have separate targets for `intel` and `nvidia`.

```suggestion
CC=dpcpp
CFLAGS=-O2 -Wall
LDFLAGS=
INCLUDES=-Iincludes

intel: CC = dpcpp

nvidia: CC = clang++
nvidia: CFLAGS += -fsycl -fsycl-targets=nvptx64-nvidia-cuda

all: intel nvidia

intel: main.c dw_cuda_soa.dp.cpp read_in.c
	$(CC) $(INCLUDES) $(CFLAGS) $^ -o main
	
nvidia: main.c dw_cuda_soa.dp.cpp read_in.c
	$(CC) $(INCLUDES) $(CFLAGS) $^ -o main	
```

---

I think it's best to write a CMakelists.txt that will handle this. Is it important to get to that before March 31, 2022?